src/rust/cryptography-x509-verification/src/policy/extension.rs

```diff
@@ -304,7 +304,11 @@ pub(crate) mod ee {
         };
 
         let san: SubjectAlternativeName<'_> = extn.value()?;
-        if !policy.subject.matches(&san) {
+        if !policy
+            .subject
+            .as_ref()
+            .map_or_else(|| false, |sub| sub.matches(&san))
+        {
```

Comment:
NB: This fails closed if the surrounding `Policy` doesn't have a configured subject. AFAICT this is the only reasonable behavior, since the SAN extension verifier here only makes sense when a subject is supplied by the user.

(In the context of a new `ClientVerifier` API, we'll need a different SAN extension verifier.)